ui/src/app/route-reuse.service.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  DetachedRouteHandle,
+  RouteReuseStrategy,
+  Routes
+} from '@angular/router';
+
+// Configure reuse of particular routes so components do not have to re-load
+@Injectable()
+export class RouteReuse implements RouteReuseStrategy {
+
+  public static cached: { [key: string]: any } = {};
```

Comment:
A timed cache expiration would be nice here. A simple `setTimeout()` might work well actually.

Reply:
Do you mean do this instead of the `needsRefresh` on `JobListView` so that after some amount of time going back to this component totally recreates it?

I think I kind of prefer the current strategy of never blowing away the cache so that you see something on the page immediately always (except first load). Instead maybe we could change `needsRefresh` to a timestamp and after some amount of time trigger a refresh of the page (with the loading indicator over the list). What do you think?

Reply:
My second concern here was around this cache growing indefinitely. So at first I was going to suggest an LRU cache + expiry similar to what you're describing. Then I thought maybe kill two birds with one stone with a timed expiry, since there's a limit on how quickly a person can generate queries (I'm not sure whether this would be a real problem even for very long-lived tabs). Simply for expiry handling, agreed that the approach you describe can work.